org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/PluginImageHelper.java

```diff
@@ -167,6 +174,10 @@ public ImageDescriptor getImageDescriptor(Image image) {
 			if(entry.getValue().equals(image))
 				return entry.getKey();
 		}
+		if (!displayDisposedRunnableInitialized  ) {
+			Display.getCurrent().disposeExec(()->this.dispose());
+			displayDisposedRunnableInitialized = true;
+		}
 		ImageDescriptor newDescriptor = ImageDescriptor.createFromImage(image);
 		registry.put(newDescriptor, image);
```

Comment:
I would put this in a private method like `addToRegistry` which also registers the dispose runnable

Reply:
done